### PR TITLE
Fix/artifacts copying

### PR DESCRIPTION
**Pull request by petrs143**
Unfortunately, the output path provided from AGP into GetApkArtifactTask was `build/intermediates/apk..` but the subsequent task requires `build/outputs/apk`. It causes an error during build because this subsequent task couldn't find the artifact files. 

So I had to change the task registration and find the assemble/bundle AGP task from all already registered tasks.

### CODE REVIEW

File: build-gradle-plugin/src/main/kotlin/cz/ackee/gradle/util/VariantExt.kt [hunk 1]
```diff
@@ -6,8 +6,8 @@ import org.gradle.api.Task
 import org.gradle.configurationcache.extensions.capitalized
 
 /**
- * TODO: Replace this with some more official solution how to get assemble task from variant,
- *  but for now I wasn't able to find it.
+ * TODO: Replace this with `wiredWith` function but for APK and Bundle artifacts it provides
+ *  wrong output path
```
Comment (mottljan) [inline build-gradle-plugin/src/main/kotlin/cz/ackee/gradle/util/VariantExt.kt:10]: This wording is a little confusing I think. What about something like "Replace this with `wiredWith` function once possible AGP bug fixed. Currently `SingleArtifact.APK` provides an incorrect input file path." Or something like that.

File: build-gradle-plugin/src/main/kotlin/cz/ackee/gradle/util/VariantExt.kt [hunk 2]
```diff
@@ -17,3 +17,16 @@ fun ApplicationVariant.assembleTask(project: Project, onAssembleTaskRegister: (T
         }
     }
 }
+
+/**
+ * TODO: Replace this with `wiredWith` function but for APK and Bundle artifacts it provides
+ *  wrong output path
+ */
```
Comment (mottljan) [inline build-gradle-plugin/src/main/kotlin/cz/ackee/gradle/util/VariantExt.kt:24]: same as above but for bundle artifact

File: build-gradle-plugin/src/main/kotlin/cz/ackee/gradle/util/VariantExt.kt (hunk 1 above)
Comment (petrs143) [reply build-gradle-plugin/src/main/kotlin/cz/ackee/gradle/util/VariantExt.kt]: amended

File: build-gradle-plugin/src/main/kotlin/cz/ackee/gradle/util/VariantExt.kt (hunk 2 above)
Comment (petrs143) [reply build-gradle-plugin/src/main/kotlin/cz/ackee/gradle/util/VariantExt.kt:24]: amended

**Review by mottljan — APPROVED** [review]

**Pull request merged**